docs/docs/concepts/advanced/private_execution_environment.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Private Execution Environment (PXE)
+---
+
+The Private Execution Environment (or PXE, pronounced 'pixie') is a client-side library for the execution of private operations. It is a TypeScript library and can be run within Node, such as when you run the sandbox, within the browser, or any other environment in which TypeScript can run. For example, in future it could be run inside wallet software.
+
+The PXE generates proofs of private function execution, and sends these proofs along with public function requests to the sequencer. Private inputs never leave the client-side PXE.
+
+```mermaid
+graph TD;
+
+    subgraph client[Client]
+        subgraph pxe [PXE]
+            acirSim[ACIR Simulator]
+            db[Database]
+            keyStore[KeyStore]
+        end
+    end
+
+    subgraph server[Server]
+        subgraph pxeService [PXE Service]
+            acctMgmt[Account Management]
+            contractTxInteract[Contract & Transaction Interactions]
+            noteMgmt[Note Management]
+        end
+    end
+
+    pxe -->|interfaces| server
+
+```
+
+## PXE Service 
+
+The PXE is a client-side interface of the PXE Service, which is a set of server-side APIs for interacting with the network. It provides functions for account management, contract and transaction interactions, note management, and more. For a more extensive list of operations, refer to the [PXE reference](../../apis/pxe/index.md).
+
+## Components
+
+### ACIR simulator
+
+The ACIR (Abstract Circuit Intermediate Representation) simulator generates the partial witness and the public inputs of smart contract functions. You can learn more about what it does in [this section](./acir_simulator.md).
+
+### Database
+
+The database stores transactional data and notes within the user's PXE. In the Aztec protocol, the database is implemented as a key-value database backed by LMDB. There is an interface ([GitHub](https://github.com/AztecProtocol/aztec-packages/blob/ca8b5d9dbff8d8062dbf1cb1bd39d93a4a636e86/yarn-project/pxe/src/database/pxe_database.ts)) for this PXE database that can be implemented in other ways, such as an in-memory database that can be used for testing.
+
+The database stores various types of data, including:
+
+- **Notes**: Encrypted representations of assets 
+- **Deferred Notes**: Notes that are intended for a user but cannot yet be decoded due to the associated contract not being present in the database
```

Comment:
Do you have examples of deferred notes? This is the first I've heard of them

Reply:
does new info help?
- **Deferred Notes**: Notes that are intended for a user but cannot yet be decoded due to the associated contract not being present in the database. When new contracts are deployed, there may be some time because it is accessible from the PXE database. When the PXE database is updated, deferred note are decoded.

Reply:
...may be some time ~~because~~ before(?) it is accessible...

Yes this is helpful